Client/mods/deathmatch/logic/CClientIMG.cpp

```diff
@@ -0,0 +1,245 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *               (Shared logic for modifications)
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/CClientIMG.cpp
+ *  PURPOSE:     IMG container class
+ *
+ *****************************************************************************/
+
+#include <StdInc.h>
+
+#define INVALID_ARCHIVE_ID 0xFF
+
+struct tImgHeader
+{
+    char         szMagic[4];
+    unsigned int uiFilesCount;
+};
+
+CClientIMG::CClientIMG(class CClientManager* pManager, ElementID ID) :
+    ClassInit(this),
+    CClientEntity(ID),
+    m_pImgManager(pManager->GetIMGManager()),
+    m_ucArchiveID(INVALID_ARCHIVE_ID),
+    m_usRequiredBufferSize(0)
+{
+    m_pManager = pManager;
+    SetTypeName("img");
+    m_pImgManager->AddToList(this);
+}
+
+CClientIMG::~CClientIMG()
+{
+    m_pImgManager->RemoveFromList(this);
+    if (IsStreamed())
+        StreamDisable();
+}
+
+bool CClientIMG::Load(fs::path filePath)
+{
+    if (!m_fileInfos.empty())
+        return false;
+
+    if (m_ifs.is_open())
+        return false;
+
+    if (filePath.empty())
+        return false;
+
+    if (!fs::exists(filePath))
+        return false;
+
+    m_filePath = filePath.string();
+    m_ifs = std::ifstream(filePath, std::ios::binary);
+
+/*    g_pClientGame->GetResourceManager()->ValidateResourceFile(m_strFilename, nullptr, 0);
+    if (!g_pCore->GetNetwork()->CheckFile("img", m_strFilename))
+        return false;
+*/
+
+    // Open the file
+    if (m_ifs.fail())
+    {
+        m_ifs.close();
+        return false;
+    }
+
+    tImgHeader fileHeader;
+
+    // Read header
+    m_ifs.read(reinterpret_cast<char*>(&fileHeader), sizeof(tImgHeader));
+
+    if (m_ifs.fail() || m_ifs.eof() || memcmp(&fileHeader.szMagic, "VER2", 4) != 0)
+    {
+        m_ifs.close();
+        return false;
+    }
+
+    // Read content info
+    try
+    {
+        m_fileInfos.resize(fileHeader.uiFilesCount);
+    }
+    catch (const std::bad_alloc&)
+    {
+        m_ifs.close();
+        return false;
+    }
+
+    m_ifs.read(reinterpret_cast<char*>(m_fileInfos.data()), sizeof(tImgFileInfo) * (std::streampos)fileHeader.uiFilesCount);
+    if (m_ifs.fail() || m_ifs.eof())
+    {
+        m_ifs.close();
+        return false;
+    }
+
```

Comment:
Okay, here we need to call netc to check all binary files.
Here's the code to do it:
`if (!g_pCore->GetNetwork()->CheckFile(<file type>, <filename>, buffer.data(), buffer.size()))`
The types possible are:
- dff
- col
- txd

You will have to load the whole files into a buffer. I'd recommend you use  `std::unique_ptr<char[]>(new char[m_usRequiredBufferSize])`. (Note the usage of `new` instead of `make_unique`, this is because we don't want to value initialize the array (eg.: zero it out))

Reply:
IMG file uses `FILE_SHARE_READ` asses. I don't know, but maybe it's enough

Reply:
Another checks added

Reply:
Resolved?